Dockerfile

```diff
@@ -25,26 +25,30 @@ COPY --from=frontend-builder /app/.output/public ./app/api/static/public
 RUN CGO_ENABLED=0 GOOS=linux go build \
     -ldflags "-s -w -X main.commit=$COMMIT -X main.buildTime=$BUILD_TIME -X main.version=$VERSION"  \
     -o /go/bin/api \
-    -v ./app/api/*.go
+    -v ./app/api/*.go && \
+    chmod +x /go/bin/api && \
+    # create a directory so that we can copy it in the next stage
+    mkdir /data
 
 # Production Stage
-FROM alpine:latest
+FROM gcr.io/distroless/static
 
 ENV HBOX_MODE=production
 ENV HBOX_STORAGE_DATA=/data/
 ENV HBOX_STORAGE_SQLITE_URL=/data/homebox.db?_fk=1
 
-RUN apk --no-cache add ca-certificates
-RUN mkdir /app
-COPY --from=builder /go/bin/api /app
-
-RUN chmod +x /app/api
+# Copy the binary and the (empty) /data dir and
+# change the ownership to the low-privileged user
+COPY --from=builder --chown=nonroot /go/bin/api /app/api
+COPY --from=builder --chown=nonroot /data /data
 
 LABEL Name=homebox Version=0.0.1
 LABEL org.opencontainers.image.source="https://github.com/hay-kot/homebox"
 EXPOSE 7745
 WORKDIR /app
 VOLUME [ "/data" ]
 
+# Drop root and run as low-privileged user
+USER nonroot

```

Comment:
So this would break every install that use a mounted volume correct?

Reply:
Yes, correct. This would break the setup for those who use `-v /path/data:/data` or equivalent constructs.

The fix is easy, but it is a breaking change nevertheless. In fact, I think this is the most useful although the most controversial change, as it is a prioritization of security over UX. 

I updated the documentation to reflect the requirement to `chown` the path accordingly.